MD5 symbols missing from Windows library

### Version

v18.12.1

### Platform

Microsoft Windows NT 10.0.22000.0 x64

### Subsystem

_No response_

### What steps will reproduce the bug?

1. Run `node-gyp rebuild` on Windows

md5.c:

```c
#include <openssl/md5.h>

void hash(void)
{
        MD5_CTX c;
        MD5_Init(&c);
        MD5_Update(&c, NULL, 0);
        MD5_Final(NULL, &c);
}
```

binding.gyp:

```python
{ 'targets': [{ 'target_name': 'md5', 'sources': ['md5.c'] }] }
```

### How often does it reproduce? Is there a required condition?

Always

### What is the expected behavior?

Builds successfully as it does on other platforms (and Node.js 16 on Windows).

### What do you see instead?

```
md5.obj : error LNK2001: unresolved external symbol MD5_Update [C:\Users\User\Development\md5\build\md5.vcxproj]
md5.obj : error LNK2001: unresolved external symbol MD5_Final [C:\Users\User\Development\md5\build\md5.vcxproj]
md5.obj : error LNK2001: unresolved external symbol MD5_Init [C:\Users\User\Development\md5\build\md5.vcxproj]
```

### Additional information

_No response_

--- a/node.gyp
+++ b/node.gyp
@@ -823,7 +823,8 @@
               '-CAES,BF,BIO,DES,DH,DSA,EC,ECDH,ECDSA,ENGINE,EVP,HMAC,MD4,MD5,'
               'PSK,RC2,RC4,RSA,SHA,SHA0,SHA1,SHA256,SHA512,SOCK,STDIO,TLSEXT,'
               'UI,FP_API,TLS1_METHOD,TLS1_1_METHOD,TLS1_2_METHOD,SCRYPT,OCSP,'
-              'NEXTPROTONEG,RMD160,CAST,DEPRECATEDIN_1_1_0,DEPRECATEDIN_1_2_0',
+              'NEXTPROTONEG,RMD160,CAST,DEPRECATEDIN_1_1_0,DEPRECATEDIN_1_2_0,'
+              'DEPRECATEDIN_3_0',
               # Defines.
               '-DWIN32',
               # Symbols to filter from the export list.

--- a/test/addons/openssl-binding/binding.cc
+++ b/test/addons/openssl-binding/binding.cc
@@ -1,8 +1,8 @@
+#include <assert.h>
+#include <node.h>
 #include <openssl/md5.h>
 #include <openssl/rand.h>
 #include <openssl/ssl.h>
-#include <node.h>
-#include <assert.h>
 
 namespace {
 

--- a/test/addons/openssl-binding/binding.cc
+++ b/test/addons/openssl-binding/binding.cc
@@ -1,3 +1,4 @@
+#include <openssl/md5.h>
 #include <openssl/rand.h>
 #include <openssl/ssl.h>
 #include <node.h>
@@ -19,6 +20,21 @@ inline void RandomBytes(const v8::FunctionCallbackInfo<v8::Value>& info) {
   info.GetReturnValue().Set(rval > 0);
 }
 
+inline void Hash(const v8::FunctionCallbackInfo<v8::Value>& info) {
+  assert(info[0]->IsArrayBufferView());
+  auto view = info[0].As<v8::ArrayBufferView>();
+  auto byte_offset = view->ByteOffset();
+  auto len = view->ByteLength();
+  assert(view->HasBuffer());
+  auto buffer = view->Buffer();
+  auto contents = buffer->GetBackingStore();
+  auto data = static_cast<unsigned char*>(contents->Data()) + byte_offset;
+  unsigned char md[MD5_DIGEST_LENGTH];
+  MD5_CTX c;
+  auto rval = MD5_Init(&c) && MD5_Update(&c, data, len) && MD5_Final(md, &c);
+  info.GetReturnValue().Set(rval > 0);
+}
+
 inline void Initialize(v8::Local<v8::Object> exports,
                        v8::Local<v8::Value> module,
                        v8::Local<v8::Context> context) {
@@ -32,6 +48,12 @@ inline void Initialize(v8::Local<v8::Object> exports,
 
   const SSL_METHOD* method = TLSv1_2_server_method();
   assert(method != nullptr);
+
+  key = v8::String::NewFromUtf8(isolate, "hash").ToLocalChecked();
+  value = v8::FunctionTemplate::New(isolate, Hash)
+              ->GetFunction(context)
+              .ToLocalChecked();
+  assert(exports->Set(context, key, value).IsJust());
 }
 
 }  // anonymous namespace

--- a/test/addons/openssl-binding/test.js
+++ b/test/addons/openssl-binding/test.js
@@ -9,3 +9,4 @@ const binding = require(`./build/${common.buildType}/binding`);
 const bytes = new Uint8Array(1024);
 assert(binding.randomBytes(bytes));
 assert(bytes.reduce((v, a) => v + a) > 0);
+assert(binding.hash(bytes));

